server/controllers/chartsController/index.js

```diff
@@ -1,80 +1,156 @@
+    areFiltersRequested,
+    filters,
+  })
   const allowedStudies = userAccess.filter(
     (study) => !STUDIES_TO_OMIT.includes(study)
   )
   const studyTotals = generateStudyTargetTotals(chart, allowedStudies)
 
-  for await (const subject of allSubjects) {
-    const { study } = subject
-    const subjectDayData = await dataDb
-      .collection(subject.collection)
-      .find({})
-      .toArray()
+  if (allSubjects.length) {
+    for await (const subject of allSubjects) {
+      const { study } = subject
+      const subjectDayData = await dataDb
+        .collection(subject.collection)
+        .find({})
+        .toArray()
 
-    chart.fieldLabelValueMap.forEach((fieldLabelValueMap) => {
-      const { color, label, value, targetValues } = fieldLabelValueMap
-      const targetValue = targetValues[study]
-      const siteName = SITE_NAMES[study] || study
-      const dataKey = `${siteName}-${label}-${targetValue}`
-      const totalsDataKey = `${TOTALS_STUDY}-${label}`
-      const isVariableValueEmpty = value === ''
-      const shouldCountSubject = isVariableValueEmpty
-        ? subjectDayData.every((day) => day[chart.variable] === value)
-        : subjectDayData.some((dayData) => dayData[chart.variable] == value)
+      chart.fieldLabelValueMap.forEach((fieldLabelValueMap) => {
+        const { color, label, value, targetValues } = fieldLabelValueMap
+        const targetValue = targetValues[study]
+        const siteName = SITE_NAMES[study] || study
+        const dataKey = `${siteName}-${label}-${targetValue}`
+        const totalsDataKey = `${TOTALS_STUDY}-${label}`
+        const isVariableValueEmpty = value === EMPTY_VALUE
+        const shouldCountSubject = isVariableValueEmpty
+          ? subjectDayData.every((day) => day[chart.variable] === value)
+          : subjectDayData.some((dayData) => dayData[chart.variable] == value)
 
-      labelMap.set(label, { name: label, color })
-      processData({ shouldCountSubject, dataMap, dataKey, totalsDataKey })
-      processTotals({
-        shouldCountSubject,
-        studyTotals,
-        siteName,
-        targetValue,
+        labelMap.set(label, { name: label, color })
+        processData({ shouldCountSubject, dataMap, dataKey, totalsDataKey })
+        processTotals({
+          shouldCountSubject,
+          studyTotals,
+          siteName,
+          targetValue,
+        })
       })
-    })
-  }
+    }
+
+    const dataBySite = postProcessData(dataMap, studyTotals)
 
-  const dataBySite = postProcessData(dataMap, studyTotals)
+    if (isAnyTargetIncluded(studyTotals)) {
+      labelMap.set(N_A, { name: N_A, color: '#808080' })
+    }
 
-  if (isAnyTargetIncluded(studyTotals)) {
-    labelMap.set(N_A, { name: N_A, color: '#808080' })
+    return {
+      chart,
+      dataBySite: Array.from(dataBySite.values()),
+      labels: Array.from(labelMap.values()),
+      studyTotals,
+      filters,
+    }
+  } else {
+    return {
+      chart,
+      dataBySite: [],
```

Comment:
`Array.from(dataBySite.values())` when `dataBySite` is it's initial value (`new Map()`) is an empty array. Do we need to hard-code this or can we keep the return value here the same as the return value above on line 88?

Reply:
https://github.com/AMP-SCZ/dpdash/pull/248/commits/147bf8a9965b532f34bb176e1e651f339c9bf387